docs/proposals/container-runtime-interface/capabilities.md

```diff
@@ -0,0 +1,221 @@
+- Specify the `Add` and `Drop` set for the containers in a k8s pod spec.
+- Launch the pod/container.
+
+There are several issues here:
+
+- The default capability set heavily depends on Docker's implementation; Kubernetes has no control over that.
+When Docker changes the default capability set across versions, the effective set of default capabilities can change suddenly.
+- To implement the same semantics as today's Docker runtime, other runtimes (e.g. rkt) are required to depend on the implicit Docker default capability set as well. If Docker changes the default capability set, then consistency is not guaranteed.
+- From a security point of view, an explicit capability set which does not rely on undefined defaults is preferable for auditing purposes.
+
+## Solution
+
+The proposed solution will:
+
+- Keep the Kubernetes API backwards compatible.
+- Add a new `RequestedSet` field to the `PodSpec.SecurityContext.Capabilities` struct, to allow users to request a specific set of capabilities.
+- Add a new `DefaultCapabilities` field to the `PodSecurityPolicySpec` struct, to allow administrators to define a default set of capabilities.
+- Define a default capability whitelist inside the following Kubernetes components:
+  1. the *API server*: Capability defaults are being applied to `PodSpec.SecurityContext.Capabilities.RequestedSet` either through the `PodSecurityPolicySpec` or, if admission control is disabled, using a default capability whitelist.
+  2. the *Kubelet*: If the kubelet receives a `PodSpec.SecurityContext.Capabilities.RequestedSet` that does not specify any capabilities, the kubelet applies capability defaults to retain backwards compatibility. This could occur for example when a kubelet is newer than the API server.
+- Make CRI expose the whitelist based capability interface, remove the existing `add_capabilities`, and `drop_capabilities` fields, and put the burden on the runtime to implement that.
+- Make the final requested capability set **visible** through the container spec, for example:
+
+```shell
+$ kubectl get pod -o yaml
+apiVersion: v1
+items:
+- apiVersion: v1
+  kind: Pod
+  metadata:
+  ...
+  spec:
+    containers:
+    - command:
+      - sleep
+      - "1000"
+      image: busybox
+      imagePullPolicy: Always
+      name: my-busybox
+      securityContext:
+        capabilities:
+          requestedSet:
+          - SETPCAP, MKNOD, AUDIT_WRITE, ...
+          add:
+          - NET_ADMIN
+          drop:
+          - MKNOD
+      ...
+  status:
+    containerStatuses:
+    - image: busybox
+      name: my-busybox
+      state:
+        running:
+          startedAt: 2016-09-27T22:33:38Z
+    ...
+```
+
+## Struct changes
+
+The `PodSecurityPolicySpec` in `pkg/apis/extensions/types.go` gains a new `DefaultCapabilities` field. 
+The existing fields `DefaultAddCapabilities`, and `RequiredDropCapabilities` will be marked as deprecated.
```

Comment:
I thought we [decided](https://github.com/kubernetes/kubernetes/pull/35910#discussion_r87462016) to keep `RequiredDropCapabilities`?